docs: Fundamentals (#1074)
vtex/faststore: docs/docs/tutorials/fundamentals/faststore.md

---
id: 1

> title: Faststore

Reviewer: ```suggestion
title: FastStore
```

> sidebar_label: What is Faststore

Reviewer: ```suggestion
sidebar_label: What is FastStore
```

---

> *[Faststore](https://github.com/vtex/faststore) is a  collection of **open-source** libraries consisting of a ecommerce-focused SDK package, a React UI components library and a GraphQL API to connect to your ecommerce provider.*

Reviewer: ```suggestion
*[FastStore](https://github.com/vtex/faststore) is a  collection of **open-source** libraries consisting of an ecommerce-focused SDK package, a React UI components library and a GraphQL API to connect to your ecommerce provider.*
```

---

> ## What is Faststore 

Reviewer: ```suggestion
## What is FastStore 
```

> [Faststore](https://github.com/vtex/faststore) is a light yet powerful **set of libraries** made for developers who want to explore their knowledge and creativity building comprehensive and unique ecommerce solutions. It supports integrations with headless ecommerce platforms, allows meaningful analytics capabilities, and gives developers back the freedom of customization.

Reviewer: ```suggestion
[FastStore](https://github.com/vtex/faststore) is a light yet powerful **set of libraries** made for developers who want to explore their knowledge and creativity building comprehensive and unique ecommerce solutions. It supports integrations with headless ecommerce platforms, allows meaningful analytics capabilities, and gives developers back the freedom of customization.
```

## What is included

> That is possible because Faststore was cleverly split into **a UI, SDK and API packages.** The UI package is a **CSS framework-agnostic** library that crafts the behavior of React components into data attributes. The SDK is an extensible state management library that handles all meaningful states an e-commerce store might have, such as:

Reviewer: ```suggestion
That is possible because FastStore was cleverly split into **a UI, SDK and API packages.** The UI package is a **CSS framework-agnostic** library that crafts the behavior of React components into data attributes. The SDK is an extensible state management library that handles all meaningful states an ecommerce store might have, such as:
```

- **Session** - handles auth, region, locale, and currency data.
- **Cart** - handles the addition, remotion, update, and current state of a store cart.
- **UI** - handles the state of the components of a store (e.g., display/hide mini cart).
- **Search** - handles the state of the faceted search (e.g., apply filters and sort).

> Our SDKs also provide GA4 compatible analytics functions to help you create powerful analytics capabilities in your e-commerce. 

Reviewer: ```suggestion
Our SDKs also provide GA4 compatible analytics functions to help you create powerful analytics capabilities in your ecommerce. 
```

> Finally, the API package let's you connect to your favorite ecommerce provider by creating interfaces for querying products, collections and manipulating carts

Reviewer: ```suggestion
Finally, the API package lets you connect to your favorite ecommerce provider by creating interfaces for querying products, collections and manipulating carts
```

> Faststore never constrains developers to any proprietary framework. It is opinionated by default but flexible under the hood.

Reviewer: ```suggestion
FastStore never constrains developers to any proprietary framework. It is opinionated by default but flexible under the hood.
```

> ## Our philosphy

Reviewer: ```suggestion
## Our philosophy
```

> Instead of focussing on the next big thing in ecommerce, we tend to focus on what will never change, and try to nail it really well. Our pillars are:
> 1. Performance: Amazing UX with great points on Lighthouse and Web Vitals
> 2. Stability: Your ecommerce should never loose a sale due to provider issues
> 3. Analytics/SEO: Understand who's visiting your store and reach a higher organic traffic

Reviewer: ```suggestion
Instead of focussing on the next big thing in ecommerce, we tend to focus on what will never change, and try to nail it really well. Our pillars are:
1. Performance: Amazing UX with great points on Lighthouse and Web Vitals
2. Stability: Your ecommerce should never lose a sale due to provider issues
3. Analytics/SEO: Understand who's visiting your store and reach a higher organic traffic
```
